lib/soapy_cake/admin_batched.rb

```diff
@@ -35,7 +35,8 @@ def to_enum
             result = next_batch
             # raises StopIteration when less than `limit` elements are present
             # which is then rescued by `loop`
-            limit.times { y << result.next }
+            opts[:xml_response] ? y << result.next : limit.times { y << result.next }
```

Comment:
Please don't use the ternary operator for statements that have side effects, it's hard to understand.